File: src/main/java/ankol/mod/merger/merger/scr/news/node/ScrFunCallNode.java
package ankol.mod.merger.merger.scr.news.node;

import lombok.Data;

import java.util.List;

@Data
public class ScrFunCallNode extends ScrNode {
    private final String functionName;
    private final List<String> arguments;

    public ScrFunCallNode(String signature, int startIndex, int stopIndex, int line, String sourceText, String functionName, List<String> arguments) {
        super(signature, startIndex, stopIndex, line, sourceText);
        this.functionName = functionName;
        this.arguments = arguments;
    }
}

File: src/main/java/ankol/mod/merger/mergers/scr/node/ScrFunCallNode.kt
package ankol.mod.merger.mergers.scr.node

import ankol.mod.merger.core.BaseTreeNode
import org.antlr.v4.runtime.TokenStream

/**
 * 函数调用节点
 */
class ScrFunCallNode(
    signature: String,
    startTokenIndex: Int,
    stopTokenIndex: Int,
    line: Int,
    tokenStream: TokenStream,
    val functionName: String,
    val arguments: List<String>
) : BaseTreeNode(signature, startTokenIndex, stopTokenIndex, line, tokenStream)
